lib/rucio/core/oidc.py

```diff
@@ -96,11 +96,13 @@ def _token_cache_get(
         METRICS.counter('token_cache.miss').inc()
         return None
 
-    try:
-        assert isinstance(value, str)
-        payload = JWT().unpack(value).payload()
-    except Exception:
-        METRICS.counter('token_cache.invalid').inc()
+    if isinstance(value, str):
+        try:
+            payload = JWT().unpack(value).payload()
+        except Exception:
+            METRICS.counter('token_cache.invalid').inc()
+            return None
+    else:
         return None
```

Comment:
Beforehand, `METRICS.counter('token_cache.invalid').inc()` would be called even in case of an `AssertionError` (under `except Exception`), but in the new version it will just `return None` when `isinstance(value, str)` is false.

```suggestion
    if isinstance(value, str):
        try:
            payload = JWT().unpack(value).payload()
        except Exception:
            METRICS.counter('token_cache.invalid').inc()
            return None
    else:
        METRICS.counter('token_cache.invalid').inc()
        return None
```

There might be a way to simplify this to avoid duplicating `METRICS.counter('token_cache.invalid').inc()` and `return None`, but nothing comes to mind at the moment